idx/memory/memory.go

```diff
@@ -453,28 +511,20 @@ func (m *UnpartitionedMemoryIdx) MetaTagRecordList(orgId uint32) []idx.MetaTagRe
 // indexTags reads the tags of a given metric definition and creates the
 // corresponding tag index entries to refer to it. It assumes a lock is
 // already held.
-func (m *UnpartitionedMemoryIdx) indexTags(def *schema.MetricDefinition) {
+func (m *UnpartitionedMemoryIdx) indexTags(def *idx.MetricDefinition) {
 	tags, ok := m.tags[def.OrgId]
 	if !ok {
 		tags = make(TagIndex)
 		m.tags[def.OrgId] = tags
 	}
 
-	for _, tag := range def.Tags {
-		tagSplits := strings.SplitN(tag, "=", 2)
-		if len(tagSplits) < 2 {
-			// should never happen because every tag in the index
-			// must have a valid format
-			invalidTag.Inc()
-			log.Errorf("memory-idx: Tag %q of id %q has an invalid format", tag, def.Id)
-			continue
-		}
-
-		tagName := tagSplits[0]
-		tagValue := tagSplits[1]
-		tags.addTagId(tagName, tagValue, def.Id)
+	for _, tag := range def.Tags.KeyValues {
+		tags.addTagId(tag.Key, tag.Value, def.Id)
 	}
-	tags.addTagId("name", def.Name, def.Id)
+	// TODO: add special case to handle name and intern the entire thing
```

Comment:
is that todo still supposed to be there? aren't the following 2 lines that special case?

Reply:
That comment perhaps could be more clear, what I meant to say was intern the string in pieces as opposed to just tossing the entire thing into the store